### PR TITLE
[API] Add "before" query to ListIssueComments and ListRepoIssueComments

### PR DESCRIPTION
**Pull request by 6543**
as title

### CODE REVIEW

**Comment by lunny** [comment]
It's betterto add some tests.

File: models/issue_comment.go [hunk 1]
```diff
@@ -799,6 +800,9 @@ func (opts *FindCommentsOptions) toConds() builder.Cond {
 	if opts.Since > 0 {
 		cond = cond.And(builder.Gte{"comment.updated_unix": opts.Since})
 	}
+	if opts.Before > 0 {
+		cond = cond.And(builder.Lte{"comment.updated_unix": opts.Since})
```
Comment (gary-kim) [inline models/issue_comment.go]: Was this meant to be `opts.Since`?

Comment (6543) [inline models/issue_comment.go]:
```suggestion
		cond = cond.And(builder.Lte{"comment.updated_unix": opts.Before})
```

**Comment by 6543** [comment]
@lunny done

**Comment by 6543** [comment]
@gary-kim are you ok with this PR now ;)

**Review by zeripath — APPROVED** [review]

**Review by gary-kim — APPROVED** [review]

**Review by sapk — APPROVED** [review]

**Pull request merged**